docs/migration/README.md

```diff
@@ -0,0 +1,152 @@
+in short it aims to replace the current workflows and code with much smaller workflows that use the
+tuf-on-ci actions.
+
+## Summary
+
+Roughly he migration steps are
+1. Prepare the GitHub project, workflows, cloud services and the data
+   _without disrupting the current operations_
+2. Run the initial signing event with tuf-on-ci, do the necessary changes to metadata _without disrupting
+   the current operations_: this is achieved by
+   * modifying a separate copy of the metadata
+   * disabling the metadata publishing from tuf-on-ci
+3. If CI tests pass and manual testing looks good: disable current repository workflows, enable metadata
+   publishing from tuf-on-ci
+4. Remove now obsolete workflows, code and metadata copies
+
+The complication here is that starting step 2 (making a copy of the current metadata) starts a timer as
+the online signing machinery will modify metadata every three days. Within that time we need to either:
+* Decide the signing event was not successful and restart step 2 at a later date or
+* complete steps 2 and 3
+
+## Detailed playbook
+
+### 1. Preparation
+
+All of these steps can be taken without time pressure from the online signing process.
+
+#### Update GitHub project settings
+
+Review root-signing settings in sigstore/community, compare to root-signing-staging settings
+and https://github.com/theupdateframework/tuf-on-ci/blob/main/docs/REPOSITORY-MAINTENANCE.md:
+Enable things that are required by tuf-on-ci (do not disable the permissions required by legacy root-signing).
+
+Define GitHub variables and secrets:
+* Variable GCP_SERVICE_ACCOUNT (this is the online signing service account)
+* Variable GCP_WORKLOAD_IDENTITY_PROVIDER (this is the online signing identity provider)
+* Secret TUF_ON_CI_TOKEN (this is a sigstore-bot token)
+
+#### Update GCP configuration
+
+The settings should be easy to copy from root-signing-staging, with the exception that there are now two different
+service accounts: one for KMS and one for GCS
+* KMS: Online signing must be possible from the online-sign workflow from the main branch
+* GCS: uploading must be possible from publish workflow from the publish branch
+
+#### Update targets/ directory in git
+
+Currently targets/ does not quite match the repository contents.
+
+#### Add tuf-on-ci workflows to git
+
+Add the workflows but disable scheduled runs and specifically prevent publishing before it is enabled.
+
+#### Add import configuration to git
+
+* Add import script that copies metadata in expected locations and rewrites whitespace (so future diffs are manageable)
+* Add configuration file for the actual tuf-on-ci import
+
+#### Add/Update playbooks to git
+
+This playbook and manuals for signers and maintainers.
+
+#### Signer orientation
+
+Make sure signers have installed and tested the software, and understand what will be requested during the signing event
+
+#### General outreach
+
+Communicate the upcoming changes to Sigstore community
```

Comment:
How? How much notice will you give? I know root-signing changes have been difficult to communicate in the past with typically one or more folks feeling they hadn't had notice. cc @haydentherapper

Reply:
This is a fair concern. I think the prep work has been done this time: 
* staging should now be a fairly solid example of what's going to happen
* clients are being  tested with staging (and will be with prod)
* the one expected failure (sigstore-rs) has been communicated to that project

What we can do now is
* make the timeline clear: as soon as we know the estimated date, publish it (it's currently looking like we can't manage this before august because of still missing npmjs signer support)
* keep the clients slack channel informed during the process

I'll take additional suggestions.

Reply:
Going forward, we hopefully won't have breaking changes so clients won't need to follow along.

As for getting the keyholders ready, just need to make sure we schedule regular signings with a few weeks notice. Might be worth revisiting the set of keyholders as well and make sure everyone is still onboard with being available with minimal notice in the event of needing an immediate rotation.